### DISCUSSION TITLE
[config-plugins] Help with iOS Plugin

### DISCUSSION THREAD

**Discussion by matallui** [expo/expo#18725] [Archived: Questions] [answered]
Hi! I'm trying to create a config plugin to pretty much perform all the steps in [this guide](https://jitsi.github.io/handbook/docs/dev-guide/dev-guide-ios-sdk/#screen-sharing-integration). Basically I'll have to:

 - Create a Broadcast Upload Extension (not sure how to do that)
 - Create group and put both main app and extension under same group (not sure how)
 - Add files to the extension target (I guess I can use `createBuildSourceFile()` for that?)
 - Modify the main app's Info.plist (using `withInfoPlist()`)
 
 From the documentation, the only thing I think I may be able to figure out with more or less effort is the Info.plist mods.
 As for the rest, I've been looking though the `@expo/config-plugins` code to try and figure it out, but it's still not obvious to me.
 
 Can someone point me to any extra docs? Or give me some guidance on how I can achieve the above steps?
 
 Any help is much appreciated!

**Comment by keith-kurak** [comment]
At first glance, I'm guessing many of these steps will require editing the **.xcodeproj** file, so good to look at the `withXcodeProject()` [plugin](https://docs.expo.dev/guides/config-plugins/#mod-plugins). A good first step is to look at the changes XCode makes to this file after you complete the configuration steps manually.

We just started a #creating-config-plugins channel on our [Discord](https://chat.expo.dev), you may want to check that out to get the feedback from the plugin creator community.

**Reply by matallui** [reply]
Awesome, thanks for the prompt response. I'll check out the Discord channel.

Meanwhile I just got the Extension created via config plugin. I've relied heavily on [this reference project](https://github.com/andrew-levy/react-native-safari-extension) which also has a config plugin to create a different type of extension.

I feel pretty confident now that I'll be able to complete this plugin.

**Comment by matallui** [comment, marked as answer]
I think I've managed to get it all done, or at least I'm pretty close. Closing this discussion.

For reference, here's what I came up with: https://github.com/matallui/demo-screen-capture
